src/OpenTelemetry.Contrib.Instrumentation.Elasticsearch/Implementation/ElasticsearchRequestPipelineDiagnosticListener.cs

```diff
@@ -0,0 +1,207 @@
+                var success = this.successFetcher.Fetch(payload);
+                var statusCode = this.httpStatusFetcher.Fetch(payload);
+
+                if (!statusCode.HasValue)
+                {
+                    activity.SetStatus(Status.Unknown);
+                }
+                else if (statusCode >= 200 && statusCode < 300)
+                {
+                    activity.SetStatus(Status.Ok);
+                }
+                else if (statusCode == 404)
+                {
+                    activity.SetStatus(Status.NotFound);
+                }
+                else if (statusCode == 401)
+                {
+                    activity.SetStatus(Status.Unauthenticated);
+                }
+                else
+                {
+                    activity.SetStatus(Status.Unknown);
+                }
+
+                var debugInformation = this.debugInformationFetcher.Fetch(payload);
+                if (debugInformation != null)
+                {
+                    activity.SetTag(Constants.AttributeDbStatement, this.ParseAndFormatRequest(activity, debugInformation));
+                }
+
+                var originalException = this.orginalExceptionFetcher.Fetch(payload);
+                if (originalException != null)
+                {
+                    activity.SetCustomProperty(Constants.ExceptionCustomPropertyName, originalException);
+
+                    if (originalException is HttpRequestException)
+                    {
+                        if (originalException.InnerException is SocketException exception)
+                        {
+                            switch (exception.SocketErrorCode)
+                            {
+                                case SocketError.HostNotFound:
+                                    activity.SetStatus(Status.InvalidArgument.WithDescription(originalException.Message));
+                                    return;
+                            }
+                        }
+
+                        if (originalException.InnerException != null)
+                        {
+                            activity.SetStatus(Status.Unknown.WithDescription(originalException.Message));
+                        }
+                    }
+                }
+            }
+
+            this.activitySource.Stop(activity);
+        }
+
+        private string ParseAndFormatRequest(Activity activity, string debugInformation)
+        {
+            if (!this.options.ParseAndFormatRequest)
+            {
+                return debugInformation;
+            }
+
+            var request = ParseRequest.Match(debugInformation);
+            if (request.Success)
+            {
+                string method = activity.GetTagValue(Constants.AttributeDbMethod).ToString();
+                string url = activity.GetTagValue(Constants.AttributeDbUrl).ToString();
+                string body = request.Groups[1].Value.Trim();
+
+                var doc = JsonDocument.Parse(body);
```

Comment:
Should we add a "pretty print" type of option? Only because this logic looks very expensive from a performance perspective.

Reply:
I guess you already have to opt into the parsing + formating. Sorry, nevermind 😄

Reply:
My intention is that is what the ParseAndFormatRequest option is for.